cmd/infra/powervs/service_id.go

```diff
@@ -0,0 +1,153 @@
+    include.release.openshift.io/single-node-developer: "true"
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: openshift-powervs-cloud-controller-manager
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: IBMCloudPowerVSProviderSpec
+    policies:
+    - attributes:
+      - name: resourceType
+        value: resource-group
+      roles:
+      - crn:v1:bluemix:public:iam::::role:Viewer
+    - attributes:
+      - name: serviceName
+        value: is
+      roles:
+      - crn:v1:bluemix:public:iam::::role:Editor
+      - crn:v1:bluemix:public:iam::::role:Operator
+      - crn:v1:bluemix:public:iam::::role:Viewer
+    - attributes:
+      - name: serviceName
+        value: power-iaas
+      roles:
+      - crn:v1:bluemix:public:iam::::role:Viewer
+      - crn:v1:bluemix:public:iam::::serviceRole:Reader
+      - crn:v1:bluemix:public:iam::::serviceRole:Manager
+`
+var nodePoolManagementCR = `
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-capi-powervs
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: IBMCloudPowerVSProviderSpec
+    policies:
+    - attributes:
+      - name: serviceName
+        value: power-iaas
+      roles:
+      - crn:v1:bluemix:public:iam::::serviceRole:Manager
+      - crn:v1:bluemix:public:iam::::role:Editor
+`
+
+var ingressOperatorCR = `
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-ingress-powervs
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: IBMCloudPowerVSProviderSpec
+    policies:
+    - attributes:
+      - name: serviceName
+        value: internet-svcs
+      roles:
+      - crn:v1:bluemix:public:iam::::serviceRole:Manager
+      - crn:v1:bluemix:public:iam::::role:Editor
+`
+
+const (
+	KubeCloudControllerManager = "kubeCloudControllerManager"
+	NodePoolManagement         = "nodePoolManagement"
+	IngressOperator            = "ingressOperator"
+)
+
+var SecretsToManage = map[string]string{
+	KubeCloudControllerManager: kubeCloudControllerManagerCR,
+	NodePoolManagement:         nodePoolManagementCR,
+	IngressOperator:            ingressOperatorCR,
+}
+
+// createServiceIDClient creates cloud credential operator's serviceID client
+func createServiceIDClient(name, APIKey, accountID, resourceGroupID, crYaml, secretRefName, secretRefNamespace string) (*cco.ServiceID, error) {
+	ccoIBMClient, err := ccoibmcloud.NewClient(APIKey, &ccoibmcloud.ClientParams{InfraName: name})
+	if err != nil {
+		return nil, err
+	}
+
+	cr := &credreqv1.CredentialsRequest{}
+	decoder := yaml.NewYAMLOrJSONDecoder(strings.NewReader(crYaml), 4096)
+	cr.Spec.SecretRef = corev1.ObjectReference{Name: secretRefName, Namespace: secretRefNamespace}
```

Comment:
can this be set post `decoder.Decode(cr)` call so that accidentally this value don't get overwritten.

Reply:
yes, should be after Decode() call, modified it, thanks!